core/src/main/java/org/fao/geonet/api/records/attachments/CMISStore.java

```diff
@@ -0,0 +1,618 @@
+        final int metadataId = canDownload(context, metadataUuid, visibility, approved);
+        final SettingManager settingManager = context.getBean(SettingManager.class);
+
+        final String resourceTypeDir = getMetadataDir(context, metadataId) + CMISConfiguration.getFolderDelimiter() + visibility.toString();
+
+        List<MetadataResource> resourceList = new ArrayList<>();
+        if (filter == null) {
+            filter = FilesystemStore.DEFAULT_FILTER;
+        }
+
+        PathMatcher matcher =
+                FileSystems.getDefault().getPathMatcher("glob:" + filter);
+
+        OperationContext operationContext = CMISConfiguration.getClient().createOperationContext();
+
+        try {
+            Folder parentFolder = (Folder) CMISConfiguration.getClient().getObjectByPath(resourceTypeDir);
+
+            Map<String, Document> documentMap = getCmisObjectMap(parentFolder, null);
+            for (Map.Entry<String, Document> entry : documentMap.entrySet()) {
+                Document object = entry.getValue();
+                String cmisFilePath = entry.getKey();
+                // Only add to the list if it is a document and it matches the filter.
+                if (object instanceof Document) {
+                    Path keyPath = new File(cmisFilePath).toPath().getFileName();
+                    if (matcher.matches(keyPath)) {
+                        final String filename = getFilename(cmisFilePath);
+                        MetadataResource resource = createResourceDescription(context, settingManager, metadataUuid, visibility, filename, object.getContentStreamLength(),
+                                object.getLastModificationDate().getTime(), object.getVersionLabel(), metadataId);
+                        resourceList.add(resource);
+                    }
+                }
+            }
+        } catch (CmisObjectNotFoundException e) {
+            // ignore as it means that there is not data to list.
+        }
+
+
+        resourceList.sort(MetadataResourceVisibility.sortByFileName);
+
+        return resourceList;
+    }
+
+    private MetadataResource createResourceDescription(final ServiceContext context, final SettingManager settingManager, final String metadataUuid,
+                                                       final MetadataResourceVisibility visibility, final String resourceId, long size, Date lastModification, String version, int metadataId) {
+        String filename = getFilename(metadataUuid, resourceId);
+
+        String versionValue = null;
+        if (CMISConfiguration.isVersioningEnabled()) {
+            versionValue = version;
+        }
+
+        MetadataResource.ExternalResourceManagementProperties externalResourceManagementProperties = getExternalResourceManagementProperties(context, metadataId, metadataUuid, visibility, resourceId, filename, version);
```

Comment:
Review long lines like this one, to add breaklines.

Reply:
I'm not sure what the max length should be?  I changed my settings to wrap at 200 and it wrapped that line.

If 200 is the suggested length then I suggest adding `max_line_length = 200` to the `.editorconfig` file.